NuRadioReco/modules/channelGalacticNoiseAdder.py

```diff
@@ -91,132 +188,167 @@ def run(
-                S = (2. * (scipy.constants.Boltzmann * units.joule / units.kelvin) * freqs[passband_filter]**2 / (scipy.constants.c * units.m / units.s)**2 * (noise_temperature) * solid_angle)
-                S[np.isnan(S)] = 0
-                # calculate radiance per energy bin
-                S_per_bin = S * d_f
-                flux_sum += S_per_bin
-                # calculate electric field per energy bin from the radiance per bin
-                E = np.sqrt(S_per_bin / (scipy.constants.c * units.m / units.s * scipy.constants.epsilon_0 * (units.coulomb / units.V / units.m))) / (d_f)
-                if self.__debug:
-                    ax_1.scatter(self.__interpolaiton_frequencies / units.MHz, noise_temperatures[:, i_pixel] / units.kelvin, c='k', alpha=.01)
-                    ax_1.plot(freqs[passband_filter] / units.MHz, noise_temperature, c='k', alpha=.02)
-                    ax_2.plot(freqs[passband_filter] / units.MHz, S_per_bin / d_f / (units.watt / units.m**2 / units.MHz), c='k', alpha=.02)
-                    ax_3.plot(freqs[passband_filter] / units.MHz, E / (units.V / units.m), c='k', alpha=.02)
-
-                # assign random phases and polarizations to electric field
-                noise_spectrum = np.zeros((3, freqs.shape[0]), dtype=complex)
-                phases = np.random.uniform(0, 2. * np.pi, len(S))
-                polarizations = np.random.uniform(0, 2. * np.pi, len(S))
-
-                noise_spectrum[1][passband_filter] = np.exp(1j * phases) * E * np.cos(polarizations)
-                noise_spectrum[2][passband_filter] = np.exp(1j * phases) * E * np.sin(polarizations)
-                efield_sum += noise_spectrum
-                antenna_orientation = detector.get_antenna_orientation(station.get_id(), channel.get_id())
-                # consider signal reflection at ice surface
+            channel_spectra[channel.get_id()] = channel.get_frequency_spectrum()
+
+        for i_pixel in range(healpy.pixelfunc.nside2npix(self.__n_side)):
+            azimuth = local_coordinates[i_pixel].az.rad
+            zenith = np.pi / 2. - local_coordinates[i_pixel].alt.rad
+
+            if zenith > 90. * units.deg:
+                continue
+
+            # consider signal reflection at ice surface
+            t_theta = geometryUtilities.get_fresnel_t_p(zenith, n_ice, 1)
+            t_phi = geometryUtilities.get_fresnel_t_s(zenith, n_ice, 1)
+            fresnel_zenith = geometryUtilities.get_fresnel_angle(zenith, n_ice, 1.)
+
+            if fresnel_zenith is None:
+                continue
+
+            temperature_interpolator = scipy.interpolate.interp1d(self.__interpolation_frequencies,
+                                                                  np.log10(self.__noise_temperatures[:, i_pixel]),
+                                                                  kind='quadratic')
+            noise_temperature = np.power(10, temperature_interpolator(freqs[passband_filter]))
```

Comment:
I benchmarked a couple of interpolation methods. We could probably speed up this by a factor of ~ 2 if we would cache the interpolation functions. However, why dont we cache the `e_field_amplitude` for a given vector of frequencies (after interpolation), this could also only be optional as it would increase the time spend in `begin()` but it should make running it faster (you could also actually cache it for a given pixel and frequency vector). And it might makes the the frequency interpolation unnecessary!

Reply:
here is the benchmarking:
```
In [13]: %timeit np.interp(x, x0, y0)
4.52 µs ± 17.6 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [14]: %timeit scipy.interpolate.interp1d(x0, y0)(x)
16.1 µs ± 53.9 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [15]: %timeit scipy.interpolate.interp1d(x0, y0, kind="quadratic")(x)
55 µs ± 77.1 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)

In [16]: f = scipy.interpolate.interp1d(x0, y0, kind="quadratic")

In [17]: %timeit f(x)
21.4 µs ± 61.9 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)

In [18]: f_lin = scipy.interpolate.interp1d(x0, y0)

In [19]: %timeit f_lin(x)
8.9 µs ± 22.3 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```